src/librustc_mir/build/scope.rs

```diff
@@ -288,14 +289,19 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             if let Some(ref free_data) = scope.free {
                 let next = self.cfg.start_new_block();
                 let free = build_free(self.hir.tcx(), tmp.clone(), free_data, next);
-                self.cfg.terminate(block, free);
+                self.cfg.terminate(block, scope.id, span, free);
                 block = next;
             }
             self.scope_auxiliary[scope.id.index()]
                 .postdoms
                 .push(self.cfg.current_location(block));
         }
-        self.cfg.terminate(block, TerminatorKind::Goto { target: target });
+
+        let scope_id = self.innermost_scope_id();
```

Comment:
I suspect this might be not correct. AFAIR the loop above does not pop any scopes, therefore this will be the scope_id of the inner-most scope. I’m not sure if that’s the real intention.